File: Binary_Search/Binary_Search.rb

# Binary Search function

def binarySearch(array, searchVariable, size)

  left = 0
  right = size - 1

  while left <= right

    middle = left + (right - left) / 2

    if(array[middle] == searchVariable)
      return middle

    elsif(searchVariable < array[middle])
      right = middle - 1;

    else
      left = middle + 1;

    end

  end

  return -1

end

# Hard Coded array
array = [1, 2, 3, 4, 5]

# Searching for 1 in the array
if binarySearch(array, 1, array.length()) != -1
  puts("Element found.")
else
  puts("Element not found.")
end
# Gives Output Element found.

# Searching for 0 in the array
if binarySearch(array, 0, array.length()) != -1
  puts("Element found.")
else
  puts("Element not found.")
end
# Gives output Element not found.

File: Binary_Search/Binary_Search.php
<?php

$list = array(3, 4, 25, 45, 78);
$element_to_be_found = 25;

function binary_search($element_to_be_found, $list, $left, $right)
{
    if ($left > $right)
        return -1;

    $mid = ($right + $left) / 2;

    if($list[$mid] == $element_to_be_found)
        return $mid + 1;
    else if($list[$mid] > $element_to_be_found)
        return binary_search($element_to_be_found,$list,$left,$mid-1);
    else if($list[$mid] < $element_to_be_found)
        return binary_search($element_to_be_found,$list,$mid+1,$right);
}

$ans = binary_search($element_to_be_found, $list, 0, count($list) - 1);

if($ans == -1)
    echo "Element not found";
else
    echo "Element found at position $ans";
    
/*

Output : Element found at position 3
    
*/

?>
    
